src/m365/tenant/commands/auditlog/auditlog-report.ts

```diff
@@ -0,0 +1,272 @@
+  }
+
+  private getBatchedPromises(AuditContentLists: AuditContentList[], batchSize: number): void {
+    for (let i: number = 0; i < AuditContentLists.length; i += batchSize) {
+      const PromiseRequestBatch: Promise<AuditlogReport[]>[] = AuditContentLists.slice(i, i + batchSize < AuditContentLists.length ? i + batchSize : AuditContentLists.length).map((AuditContentList: AuditContentList) => {
+        return this.getAuditLogReportforSingleContentURL(AuditContentList.contentUri);
+      })
+      this.batchedPromises.push(PromiseRequestBatch);
+    }
+  }
+
+  private getBatchedAuditlogData(logger: Logger, batchNumber: number, resolve: () => void, reject: (err: any) => void): void {
+
+    if (this.verbose) {
+      logger.logToStderr(`Starting Batch : ${batchNumber}`);
+    }
+    Promise
+      .all(this.batchedPromises[batchNumber])
+      .then((data: any) => {
+        this.completeAuditReports.push(data);
+        if (batchNumber < this.batchedPromises.length - 1) {
+          this.getBatchedAuditlogData(logger, ++batchNumber, resolve, reject)
+        }
+        else {
+          resolve();
+        }
+      }, err => { return Promise.reject(err) });
+  }
+
+  private getAuditLogReportforSingleContentURL(auditURL: string): Promise<AuditlogReport[]> {
+    const requestOptions: any = {
+      url: auditURL,
+      headers: {
+        accept: 'application/json;'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<AuditlogReport[]>(requestOptions);
+  }
+
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-c, --contentType <contentType>',
+        description: 'Audit content type of logs to be retrieved, should be one of the following: AzureActiveDirectory, Exchange, SharePoint, General, DLP'
+      },
+      {
+        option: '-s, --startTime [startTime]',
+        description: 'Start time of logs to be retrieved. Start time and end time must both be specified (or both omitted) and must be less than or equal to 24 hours apart.'
+      },
+      {
+        option: '-e, --endTime [endTime]',
+        description: 'End time of logs to be retrieved. Start time and end time must both be specified (or both omitted) and must be less than or equal to 24 hours apart.'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(args: CommandArgs): boolean | string {
+    if ((<any>AuditContentTypes)[args.options.contentType] === undefined) {
+      return `${args.options.contentType} is not a valid value for the contentType option. Allowed values are ${Object.keys(AuditContentTypes).join(' | ')}`;
+    }
+
+    if (args.options.startTime || args.options.endTime) {
+      if (!(args.options.startTime)) {
+        return `Please specify a starttime`;
+      }
+
+      const parsedStartTime = Date.parse(args.options.startTime as string)
```

Comment:
We don't need to use `as` with `startTime`, because `undefined` has already been excluded by the check on line 230 so by now we're sure we have a string.